deps/rabbit/src/rabbit_amqp_management.erl

```diff
@@ -210,9 +210,18 @@ handle_http_req(<<"PUT">>,
             {error, not_found} ->
                 ok = prohibit_cr_lf(XNameBin),
                 ok = prohibit_reserved_amq(XName),
-                rabbit_exchange:declare(
-                  XName, XTypeAtom, Durable, AutoDelete,
-                  Internal, XArgs, Username)
+                case rabbit_exchange:declare(
+                       XName, XTypeAtom, Durable, AutoDelete,
+                       Internal, XArgs, Username) of
+                    {ok, DeclaredX} ->
+                        DeclaredX;
+                    {error, timeout} ->
+                        throw(
+                          <<"503">>,
+                          "Could not create exchange '~ts' in vhost '~ts' "
```

Comment:
Will this crash because `~s` expects a string an `XName`  is a tuple?

Reply:
Should we just use `rabbit_misc:rs/1` here?

Reply:
Ah that's a resource record. Right let's use `rabbit_misc:rs/1` for that too